app/controllers/concerns/deprecation_concern.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module DeprecationConcern
+  extend ActiveSupport::Concern
+
+  class_methods do
+    def deprecate_api(date, sunset: nil, **kwargs)
+      deprecation_timestamp = "@#{date.to_datetime.to_i}"
+      sunset = sunset&.to_date&.httpdate
+
+      before_action(**kwargs) do
+        response.headers['Deprecation'] = deprecation_timestamp
+        response.headers['Sunset'] = sunset if sunset
```

Comment:
If the date has passed, should it return an error instead? Or do you plan to handle making the removal effective elsewhere?

Reply:
I don't think it makes sense to return errors in this way, errorring out would be because we removed the deprecated controller. This was more about preparing for potentially deciding to retire endpoints in advance (up until now, we have removed very few endpoints, and we never had a real process for that)